src/components/main.jsx

```diff
@@ -41,23 +44,35 @@ const Main = () => {
 
     useEffect(() => {
         RandomNum();
-    }, [])
+        // set loading flag on render
+        setIsLoading(true);
+
+        // load location info
+        navigator.geolocation.getCurrentPosition(async position => {
+            const latitude = position.coords.latitude;
+            const longitude = position.coords.longitude;
+            setCoordsInfo({ latitude, longitude });
+            const { place, weatherTemp, clouds } = await loadData(latitude, longitude);
+            setWeather(`${place} ${weatherTemp} ${clouds}`);
+            setIsLoading(false);
+        }, err => {
+            setErrorMessage(err.message);
+            setIsLoading(false);
+        });
+    }, []);
 
-    useEffect(() => {
-        saveCoords(coordsInfo);
-        loadData();
-    }, [coordsInfo])
-
-    const loadData = () => {
-        fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${coordsInfo.latitude}&lon=${coordsInfo.longitude}&appid=${API_WEATHER_KEY}&units=metric`)
-            .then((response) => response.json())
-            .then((data) => {
-                console.log(data);
-                const weatherTemp = data.main.temp;
-                const place = data.name;
-                const clouds = data.weather[0].description;
-                return setWeather(`${place}, ${weatherTemp} ℃, ${clouds}`);
-            })
+    const loadData = async (latitude, longitude) => {
+        try {
+            const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${API_WEATHER_KEY}&units=metric`);
+            const data = await response.json();
+            const weatherTemp = data.main.temp;
+            const place = data.name;
+            const clouds = data.weather[0].description;
+            return { weatherTemp, place, clouds };
+        } catch (err) {
+            console.error(err);
+            return err;
+        }
     }
 
     const RandomNum = () => {
```

Comment:
does this need to be a separate function?

Reply:
Thank you sir.